reference.py

```diff
@@ -61,9 +61,21 @@ def get_pubkey_from_input(vin: VinInfo) -> ECPubKey:
         if (pubkey.valid) & (pubkey.compressed):
             return pubkey
     if is_p2tr(vin.prevout):
+        witnessStack = vin.txinwitness.scriptWitness.stack
+        if (len(witnessStack) > 1):
+            # Script spend
+            control_block = witnessStack[-1]
```

Comment:
I'm assuming this still works even if an optional `annex` is added? Would be really nice to also include another input (either in this test or a different test) that tests that this still works even with the annex

Reply:
@josibake Based on my understanding of https://github.com/bitcoin/bips/blob/master/bip-0341.mediawiki, the annex is optionally added as the last item of the witness and starts with a `0x50`. Therefore, I only need to update the code to check if the last item starts with `0x50`, if it does, then the control_block is `witnessStack[-2]`

Reply:
I can also update the `get_leafp2tr_witness` to add an annex byte to the witness list to complete the test

Reply:
I think my preference would be to test both cases: one with an annex, one without an annex. You can test them in the same test case, meaning you have one input that has no annex, and another input that has annex

Reply:
This has been addressed in https://github.com/S3RK/bip-352/pull/3/commits/83eb4a97f019718ba32cafe1e84366fcb6643026

Reply:
> I think my preference would be to test both cases: one with an annex, one without an annex. You can test them in the same test case, meaning you have one input that has no annex, and another input that has annex

Just saw this. I added inputs for both cases. See https://github.com/S3RK/bip-352/pull/3/commits/b4d4dd5de587be50799f75ecbe7e1f04196e893a